### DISCUSSION TITLE
Make button (links) more powerful

### DISCUSSION THREAD

**Discussion by Meldiron** [directus/directus#6089] [Feature Requests]
Button links can only open a URL for now. That is **AWESOME** functionality for buttons like "Open profile" that opens the web app with specific profile.

But... What else?

My idea is to make a new interface `Button actions` that will be able to automate processes, let me explain...

- "Publish article" that changes "status" field to "published"
- "Set out of stock" that will change "in_stock" to 0
- "Deposit 5 tokens" that will increase "token" field by +5
- "Ban user" that will change status, but also trigger some custom hook to send an e-mail to user

By implementing this, you will be forced to create a system (there maybe already is, I could not find it) for alias interface to be able to affect any fields... My alias field "publish-button" needs to be able to update the value of field "status". This can become really powerful for custom interfaces as we would be able to make even more customizable actions, such as:

- Add tokens button with number field next to it to type any amount
- Switch button that says "Ban user" if he is not banned, but "Unban user" if he is. Functionality also changes from setting status to "banned" or "active"

Do you think these use-cases are general enough? I personally think every administration can get simpler to use if we had button actions.

**Comment by rijkvanzanten** [comment]
This ties into a bigger "How to configure workflows" discussion. I'm already dreaming of a sort of no-code configurator for setting up hooks / endpoints from the GUI, in which this Button would be the "trigger" 🙌

**Comment by Dumpstar56** [comment]
This sounds like a very exciting feature! I would want to have one button that sets several Boolean fields to true. It sounds like this envisioned action button would be able to do that.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
